easybuild/easyblocks/o/openblas.py

```diff
@@ -44,3 +61,42 @@ def sanity_check_step(self):
             'dirs': [],
         }
         super(EB_OpenBLAS, self).sanity_check_step(custom_paths=custom_paths)
+    
+    def _parse_optarch( self, cfg_optarch):
+        """ Plucks the march out of the optarch settings. See
+        https://easybuild.readthedocs.io/en/latest/Controlling_compiler_optimization_flags.html """
+
+        pieces = cfg_optarch.split('=')
```

Comment:
@terjekv We already have logic to parse the `optarch` value, I don't think it's a good idea to re-implement that here...

See `_postprocess_optarch` in https://github.com/easybuilders/easybuild-framework/blob/develop/easybuild/tools/options.py#L851 .

You currently can't easily leverage that code from an easyblock though, and you can't access the parsed value either currently, I think. That can be changed in framework though (in a backwards-compatible way)

Reply:
Right, this one will take a bit more time. Another work in progress.  :-)

Reply:
Okay, so I have done some work on the framework side, but I am unable to find any actual parsing of the contents of `optarch` in the framework. There are (limited) validation tests, but nothing that does something like "extract the march from this optarch". As such, the current iteration of the EasyBlock does not rely on changes in the framework, but a future PR will look at cleaning up some of the Framework code and moving generalised code from the EB to the Framework.